spec/lib/site_settings/deprecated_settings_spec.rb

```diff
@@ -0,0 +1,223 @@
+      before do
+        deprecate_override!(
+          ["min_trust_level_to_allow_invite", "invite_allowed_groups", override, "0.0.1"],
+        )
+      end
+
+      it "uses the minimum trust level from the trust level auto groups in the new group setting" do
+        SiteSetting.invite_allowed_groups =
+          "#{Group::AUTO_GROUPS[:trust_level_3]}|#{Group::AUTO_GROUPS[:trust_level_4]}"
+        expect(SiteSetting.min_trust_level_to_allow_invite).to eq(TrustLevel[3])
+      end
+
+      it "returns nil if there are no trust level auto groups in the new group setting" do
+        SiteSetting.invite_allowed_groups = Fabricate(:group).id.to_s
+        expect(SiteSetting.min_trust_level_to_allow_invite).to eq(nil)
+      end
+
+      it "returns nil if there are only staff and admin auto groups in the new group setting" do
+        SiteSetting.invite_allowed_groups =
+          "#{Group::AUTO_GROUPS[:admins]}|#{Group::AUTO_GROUPS[:staff]}"
+        expect(SiteSetting.min_trust_level_to_allow_invite).to eq(nil)
+      end
+    end
+
+    context "when getting an old TrustLevelAndStaffSetting" do
+      before do
+        deprecate_override!(
+          [
+            "min_trust_level_to_allow_invite_tl_and_staff",
+            "invite_allowed_groups",
+            override,
+            "0.0.1",
+          ],
+          ["min_trust_level_to_allow_invite_tl_and_staff"],
+        )
+      end
+
+      it "returns staff if there are staff and admin auto groups in the new group setting" do
+        SiteSetting.invite_allowed_groups =
+          "#{Group::AUTO_GROUPS[:admins]}|#{Group::AUTO_GROUPS[:staff]}"
+        expect(SiteSetting.min_trust_level_to_allow_invite_tl_and_staff).to eq("staff")
+      end
+
+      it "returns admin if there is only the admin auto group in the new group setting" do
+        SiteSetting.invite_allowed_groups = "#{Group::AUTO_GROUPS[:admins]}"
+        expect(SiteSetting.min_trust_level_to_allow_invite_tl_and_staff).to eq("admin")
+      end
+
+      it "returns the min trust level if the admin auto group as well as lower TL auto groups in the new group setting" do
+        SiteSetting.invite_allowed_groups =
+          "#{Group::AUTO_GROUPS[:admins]}|#{Group::AUTO_GROUPS[:trust_level_3]}"
+        expect(SiteSetting.min_trust_level_to_allow_invite_tl_and_staff).to eq(TrustLevel[3])
+      end
+    end
+
+    context "when setting an old TrustLevelSetting" do
+      before do
+        deprecate_override!(
+          ["min_trust_level_to_allow_invite", "invite_allowed_groups", override, "0.0.1"],
+          ["min_trust_level_to_allow_invite"],
+        )
+      end
+
+      it "converts the provided trust level to the appropriate auto group" do
+        SiteSetting.min_trust_level_to_allow_invite = TrustLevel[4]
+        expect(SiteSetting.min_trust_level_to_allow_invite).to eq(TrustLevel[4])
+        expect(SiteSetting.invite_allowed_groups).to eq(Group::AUTO_GROUPS[:trust_level_4].to_s)
```

Comment:
I think it will be worth testing the case where `SiteSetting.invite_allowed_groups` has already been set to some groups before setting `SiteSetting.min_trust_level_to_allow_invite` where we will expect it to append the trust level groups to the existing groups.

Reply:
I don't think this should happen; if you are setting the old setting deliberately, it should override the groups for consistency.

Reply:
Ahhh OK that makes sense :+1: